### PR TITLE
BaSP-TG-15: Admins Tests

### PR DESCRIPTION
**Pull request by Valentin-sanabria**

### CODE REVIEW

File: src/tests/admins.test.js [hunk 1]
```diff
@@ -1,5 +1,156 @@
...
+  test('should return status 400', async () => {
+    const response = await request(app).get('/admins/634b30cda84415df76492ecf').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe(true);
+    expect(response.body.data).toBe(undefined);
+  });
+});
+
+describe('ERRORS PUT - invalid edits', () => {
+  test('Should return "name empty" ', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:40]:
```suggestion
  test('should return an error when the name is empty', async () => {
```

File: src/tests/admins.test.js [hunk 2]
```diff
@@ -1,5 +1,156 @@
...
+      email: 'asd@hotm.com',
+      password: '23132aaaaa',
+    };
+
+    const response = await request(app).put('/admins/634b30cda844d15df73652exd').send(fakeAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBe('There was an error: "name" is not allowed to be empty');
+  });
+
+  test('Should return "lastName empty" ', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:53]:
```suggestion
  test('should return an error when the lastName is empty', async () => {
```

File: src/tests/admins.test.js [hunk 3]
```diff
@@ -1,5 +1,156 @@
...
+      email: 'asd@hotm.com',
+      password: '23132aaaaa',
+    };
+
+    const response = await request(app).put('/admins/634b30cda844d15df73652exd').send(fakeAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBe('There was an error: "lastName" is not allowed to be empty');
+  });
+
+  test('Should return "email empty" ', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:66]:
```suggestion
  test('should return an error when the email is empty'', async () => {
```

File: src/tests/admins.test.js [hunk 4]
```diff
@@ -1,5 +1,156 @@
...
+      email: '',
+      password: '23132aaaaa',
+    };
+
+    const response = await request(app).put('/admins/634b31caa18bcb2e7eb97458').send(fakeAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBe('There was an error: "email" is not allowed to be empty');
+  });
+
+  test('Should return "password empty" ', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:79]:
```suggestion
  test('should return an error when the password is empty' ', async () => {
```

File: src/tests/admins.test.js [hunk 5]
```diff
@@ -1,5 +1,156 @@
...
+      email: 'idontexist@hot.com',
+      password: '',
+    };
+
+    const response = await request(app).put('/admins/634b31caa18bcb2e7eb97458').send(fakeAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBe('There was an error: "password" is not allowed to be empty');
+  });
+
+  test('Should return "cast to objectId failed" ', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:92]:
```suggestion
  test('Should return an error when the id is invalid', async () => {
```

File: src/tests/admins.test.js [hunk 6]
```diff
@@ -1,5 +1,156 @@
...
+    const response = await request(app).put('/admins/634b31caa18bcb2e7eb97458').send();
+    expect(response.status).toBe(400);
```
Comment (lmalvasia) [inline src/tests/admins.test.js:94]: Could you check the error message?

File: src/tests/admins.test.js [hunk 7]
```diff
@@ -1,5 +1,156 @@
...
+});
+
+describe('ERROR POST - create empty admin', () => {
+  const emptyAdmin = {
+    name: '',
+    lastName: '',
+    email: '',
+    password: '',
+  };
+  test('should return status 400', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:133]: This should be inside the previous describe.

File: src/tests/admins.test.js [hunk 8]
```diff
@@ -1,5 +1,156 @@
...
+    const response = await request(app).post('/admins').send(emptyAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe(true);
+    expect(response.body.data).toBe(undefined);
+  });
+});
+
+describe('ERROR DELETE - invalid id', () => {
+  test('should return status 400', async () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:142]:
```suggestion
  test('should return status 400 when the id is invalid', async () => {
```

Comment (lmalvasia) [reply src/tests/admins.test.js:142]: This test should be inside the other describe.

File: src/tests/admins.test.js [hunk 9]
```diff
@@ -1,5 +1,150 @@
...
+
+  test('Should return status 400 when the id is invalid', async () => {
+    const response = await request(app).get('/admins/634b30cda84415df76492ecf').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe(true);
+    expect(response.body.data).toBe(undefined);
+  });
+});
+
+describe('PUT - invalid edits', () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:37]:
```suggestion
describe('PUT - Edit admins', () => {
```

File: src/tests/admins.test.js [hunk 10]
```diff
@@ -1,5 +1,150 @@
...
+  };
+  test('should return status 400', async () => {
+    const response = await request(app).post('/admins').send(emptyAdmin);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe(true);
+    expect(response.body.data).toBe(undefined);
+  });
+});
+
+describe('DELETE tests', () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:137]:
```suggestion
describe('DELETE - Remove admins', () => {
```

File: src/tests/admins.test.js [hunk 11]
```diff
@@ -1,5 +1,150 @@
...
+import request from 'supertest';
+import app from '../app';
+import Admins from '../models/Admins';
+import adminsSeeds from '../seeds/admins';
+
+beforeAll(async () => {
+  await Admins.collection.insertMany(adminsSeeds);
+});
+
+describe('GET ALL /admins', () => {
```
Comment (lmalvasia) [inline src/tests/admins.test.js:10]:
```suggestion
describe('GET - Get admins', () => {
```

**Review by kolodani — APPROVED** [review]
great work

**Pull request merged**